ucndata/docs/settings.md
# Settings

[Ucndata Index](./README.md#ucndata-index) / Settings

> Auto-generated documentation for [settings](../../settings.py) module.

#### Attributes

- `datadir` - path to the directory which contains the root files: '/data3/ucn/root_files'

- `timezone` - timezone for datetime conversion: 'America/Vancouver'

- `cycle_times_mode` - cycle times finding mode: 'matched'

- `DET_NAMES` - detector tree names: {'He3': {'hits': 'UCNHits_He3', 'charge': 'He3_Charge', 'rate': 'He3_Rate', 'transitions': 'RunTransitions_He3', 'hitsseq': 'hitsinsequence_he3', 'hitsseqcumul': 'hitsinsequencecumul_he3'}, 'Li6': {'hits': 'UCNHits_Li-6', 'charge': 'Li6_Charge', 'rate': 'Li6_Rate', 'transitions': 'RunTransitions_Li-6', 'hitsseq': 'hitsinsequence_li6', 'hitsseqcumul': 'hitsinsequencecumul_li6'}}

- `SLOW_TREES` - needed slow control trees: for checking data quality: ('BeamlineEpics', 'SequencerTree', 'LNDDetectorTree')

- `DATA_CHECK_THRESH` - data thresholds for checking data: {'beam_min_current': 0.1, 'beam_max_current_std': 0.02, 'max_bkgd_count_rate': 4, 'min_total_counts': 100, 'pileup_cnt_per_ms': 3, 'pileup_within_first_s': 1}

- `DET_BKGD` - default detector backgrounds - from 2019: {'Li6': 1.578, 'Li6_err': 0.009, 'He3': 0.0349, 'He3_err': 0.0023}

- `tree_filter` - filter what trees and branches to load in each file. If unspecified then load the whole tree
  treename: (filter, columns). See [rootloader documentation](https://github.com/ucn-triumf/rootloader/blob/main/docs/rootloader/ttree.md#ttree-1) for details: {}


- [Settings](#settings)
  - [keyfilter](#keyfilter)

## keyfilter

[Show source in settings.py:58](../../settings.py#L58)

Don't load all the data in each file, only that which is needed

#### Signature

```python
def keyfilter(name): ...
```

ucndata/docs/ucndata/settings.md
# Settings

[Ucndata Index](../README.md#ucndata-index) / [Ucndata](./index.md#ucndata) / Settings

> Auto-generated documentation for [ucndata.settings](../../ucndata/settings.py) module.

#### Attributes

- `datadir` - path to the directory which contains the root files: '/data3/ucn/root_files'

- `timezone` - timezone for datetime conversion: 'America/Vancouver'

- `cycle_times_mode` - cycle times finding mode: 'matched'

- `DET_NAMES` - detector tree names: {'He3': {'hits': 'UCNHits_He3', 'charge': 'He3_Charge', 'rate': 'He3_Rate', 'transitions': 'RunTransitions_He3', 'hitsseq': 'hitsinsequence_he3', 'hitsseqcumul': 'hitsinsequencecumul_he3'}, 'Li6': {'hits': 'UCNHits_Li-6', 'charge': 'Li6_Charge', 'rate': 'Li6_Rate', 'transitions': 'RunTransitions_Li-6', 'hitsseq': 'hitsinsequence_li6', 'hitsseqcumul': 'hitsinsequencecumul_li6'}}

- `SLOW_TREES` - needed slow control trees: for checking data quality: ('BeamlineEpics', 'SequencerTree', 'LNDDetectorTree')

- `DATA_CHECK_THRESH` - data thresholds for checking data: {'beam_min_current': 0.1, 'beam_max_current_std': 0.02, 'max_bkgd_count_rate': 4, 'min_total_counts': 100, 'pileup_cnt_per_ms': 3, 'pileup_within_first_s': 1}

- `DET_BKGD` - default detector backgrounds - from 2019: {'Li6': 1.578, 'Li6_err': 0.009, 'He3': 0.0349, 'He3_err': 0.0023}


- [Settings](#settings)
  - [keyfilter](#keyfilter)

## keyfilter

[Show source in settings.py:49](../../ucndata/settings.py#L49)

Don't load all the data in each file, only that which is needed

#### Signature

```python
def keyfilter(name): ...
```

ucndata/ucndata/settings.py
# Overall settings. Import this, or override its values at every execution
# Derek Fujimoto
# Oct 2024

# path to the directory which contains the root files
datadir = "/data3/ucn/root_files"

# timezone for datetime conversion
timezone = 'America/Vancouver'

# cycle times finding mode
cycle_times_mode = 'matched'

# detector tree names
DET_NAMES = {'He3':{'hits':         'UCNHits_He3',
                    'charge':       'He3_Charge',
                    'rate':         'He3_Rate',
                    'transitions':  'RunTransitions_He3',
                    'hitsseq':      'hitsinsequence_he3',
                    'hitsseqcumul': 'hitsinsequencecumul_he3',
                    },
                'Li6':{'hits':         'UCNHits_Li-6',
                    'charge':       'Li6_Charge',
                    'rate':         'Li6_Rate',
                    'transitions':  'RunTransitions_Li-6',
                    'hitsseq':      'hitsinsequence_li6',
                    'hitsseqcumul': 'hitsinsequencecumul_li6',
                    },
            }

# needed slow control trees: for checking data quality
SLOW_TREES = ('BeamlineEpics', 'SequencerTree', 'LNDDetectorTree')

# data thresholds for checking data
DATA_CHECK_THRESH = {'beam_min_current': 0.1, # uA3
                     'beam_max_current_std': 0.02, # uA
                     'max_bkgd_count_rate': 4, # fractional increase over DET_BKGD values
                     'min_total_counts': 100, # number of counts total
                     'pileup_cnt_per_ms': 3, # if larger than this, then pileup and delete
                     'pileup_within_first_s': 1, # time frame for pileup in each period
                    }

# default detector backgrounds - from 2019
DET_BKGD = {'Li6':     1.578,
            'Li6_err': 0.009,
            'He3':     0.0349,
            'He3_err': 0.0023}

# filter what trees and branches to load in each file. If unspecified then load the whole tree
# treename: (filter, columns). See [rootloader documentation](https://github.com/ucn-triumf/rootloader/blob/main/docs/rootloader/ttree.md#ttree-1) for details
tree_filter = {}

# set tree_filter to read only detector entries which contain hits
for name in DET_NAMES.values():
    tree_filter[name['hits']] = ('tIsUCN>0', None)

# filter what objects to load in each file
def keyfilter(name):
    """Don't load all the data in each file, only that which is needed"""

    name = name.replace(' ', '_').lower()

    # reject some keys based on partial matches
    reject_keys = ('v1725', 'v1720', 'v792', 'tv1725', 'charge', 'edge_diff',
                   'pulse_widths', 'iv2', 'iv3', 'rate')
    for key in reject_keys:
        if key in name:
            return False

    return True

